trace/stats.go

```diff
@@ -0,0 +1,172 @@
+package trace
+
+import (
+	"sync"
+
+	"github.com/akitasoftware/akita-cli/pcap"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+// We produce a set of packet counters indexed by interface and
+// port number (*either* source or destination.)
+type PacketCounters struct {
+	// Flow
+	Interface string
+	SrcPort   int
+	DstPort   int
+
+	// Number of events
+	TCPPackets    int
+	HTTPRequests  int
+	HTTPResponses int
+	Unparsed      int
+}
+
+func (c *PacketCounters) Add(d PacketCounters) {
+	c.TCPPackets += d.TCPPackets
+	c.HTTPRequests += d.HTTPRequests
+	c.HTTPResponses += d.HTTPResponses
+	c.Unparsed += d.Unparsed
+}
+
+// A consumer accepts incremental updates in the form
+// of PacketCounters.
+type PacketCountConsumer interface {
+	// Add an additional measurement to the current count
+	Update(delta PacketCounters)
+
+	// Signal the end of the measurement interval
+	Close()
```

Comment:
Does this have any non-empty implementations?

Reply:
No, I'll remove it.

My original thinking was to use a worker goroutine to sum everything instead of doing it in a mutex-protected critical section. so I needed a Close to shut that down.  But I don't think I actually called Close so it wouldn't work anyway.

Reply:
Fixed in https://github.com/akitasoftware/akita-cli/pull/18/commits/5b903c0c1e8b5884c0cb3595f8cf212de2ae0870